### PR TITLE
Add troubleshooting guidance to OTel Operator auto-instruentation docs

### PR DESCRIPTION
**Pull request by avillela**
I've added some troubleshooting guidance for the OTel Operator's auto-instrumentation capability, based on my recent experiences with it.

### CODE REVIEW

**Review by jaronoff97 — APPROVED** [review]

**Comment by jaronoff97** [comment]
Thanks for writing these :)))))))))))))))))))))))

**Review by alolita — APPROVED** [review]
lgtm

File: content/en/docs/kubernetes/operator/automatic.md [hunk 1]
```diff
@@ -373,3 +375,175 @@ Alternatively, the annotation can be added to a namespace, which will result in
...
+by running this command:
+
+```
+kubectl logs -l app.kubernetes.io/name=opentelemetry-operator --container manager -n opentelemetry-operator-system --follow
+```
+
+### **3- Were the resources deployed in the right order?**
+
+Order matters! The `Instrumentation` resource needs to be deployed before before
+deploying the application, otherwise the auto-instrumentation won’t work.
```
Comment (TylerHelmuth) [inline content/en/docs/kubernetes/operator/automatic.md:453]: Is this true?  I thought if you had deployments with the annotation and created an Instrumentation object the operator would start its work.

File: content/en/docs/kubernetes/operator/automatic.md [hunk 2]
```diff
@@ -297,11 +297,12 @@ spec:
 EOF
 ```
 
-By default, the Instrumentation resource that auto-instruments Python services
-uses `otlp` with the `http/protobuf` protocol. This means that the configured
-endpoint must be able to receive OTLP over `http/protobuf`. Therefore, the
-example uses `http://demo-collector:4318`, which will connect to the `http` port
-of the `otlpreceiver` of the Collector created in the previous step.
+🚨 By default, the `Instrumentation` resource that auto-instruments Python
```
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:300]:
```suggestion
By default, the `Instrumentation` resource that auto-instruments Python
```

File: content/en/docs/kubernetes/operator/automatic.md [hunk 3]
```diff
@@ -373,3 +374,175 @@ Alternatively, the annotation can be added to a namespace, which will result in
 all services in that namespace to opt-in to automatic instrumentation. See the
 [Operators auto-instrumentation documentation](https://github.com/open-telemetry/opentelemetry-operator/blob/main/README.md#opentelemetry-auto-instrumentation-injection)
 for more details.
+
+## Troubleshooting
+
+If you run into problems trying to auto-instrument your code, here are a few
+things that you can try.
+
+### **1- Did the Instrumentation resource install?**
```
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:383]: - Headings are already in bold, so we don't need the extra bolding markdown syntax.
- I'd suggest dropping the numbers too.

```suggestion
### Did the Instrumentation resource install?
```

File: content/en/docs/kubernetes/operator/automatic.md [hunk 4]
```diff
@@ -373,3 +374,175 @@ Alternatively, the annotation can be added to a namespace, which will result in
...
+
+After installing the `Instrumentation` resource, verify that it installed
+correctly by running this command, where `<namespace>` is the namespace in which
+the `Instrumentation` resource is deployed:
+
+```
```
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:389]:
```suggestion
```sh
```

File: content/en/docs/kubernetes/operator/automatic.md [hunk 5]
```diff
@@ -373,3 +374,175 @@ Alternatively, the annotation can be added to a namespace, which will result in
...
+ Sampler:
+Events:  <none>
+```
+
+### **2- Do the OTel Operator logs show any auto-instrumentation errors?**
+
+Check the OTel Operator logs for any errors pertaining to auto-instrumentation
+by running this command:
+
+```
```
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:445]:
```suggestion
```sh
```

File: content/en/docs/kubernetes/operator/automatic.md [hunk 6]
```diff
@@ -373,3 +374,175 @@ Alternatively, the annotation can be added to a namespace, which will result in
...
+deploying the application, otherwise the auto-instrumentation won’t work.
+
+Recall the auto-instrumentation annotation:
+
+```yaml
+annotations:
+  instrumentation.opentelemetry.io/inject-python: 'true'
+```
+
+The above annotation tells the OTel Operator to look for an `Instrumentation`
```
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:461]:
```suggestion
The annotation above tells the OTel Operator to look for an `Instrumentation`
```

File: content/en/docs/kubernetes/operator/automatic.md [hunk 7]
```diff
@@ -373,3 +374,175 @@ Alternatively, the annotation can be added to a namespace, which will result in
...
+
+If the `Instrumentation` resource isn’t present by the time the application is
+deployed, however, the init-container can’t be created. Therefore, if the
+application is deployed _before_ deploying the `Instrumentation` resource, the
+auto-instrumentation will fail.
+
+To make sure that the `opentelemetry-auto-instrumentation` init-container has
+started up correctly (or has even started up at all), run the following command:
+
+```
```
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:480]:
```suggestion
```sh
```

File: content/en/docs/kubernetes/operator/automatic.md [hunk 8]
```diff
@@ -373,3 +374,175 @@ Alternatively, the annotation can be added to a namespace, which will result in
...
+where to send data to. This can be an [OTel Collector](/docs/collector/), or any
+OTLP endpoint. If this attribute is left out, it defaults to
+`http://localhost:4317`, which, most likely won't send telemetry data anywhere.
+
+When sending telemetry to an OTel Collector located in the same Kubernetes
+cluster, `spec.exporter.endpoint` should reference the name of the OTel
+Collector
+[`Service`](https://kubernetes.io/docs/concepts/services-networking/service/).
+
+For examples:
```
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:534]:
```suggestion
For example:
```

**Review by chalin — COMMENTED** [review]
A few more copyedits.

File: content/en/docs/kubernetes/operator/automatic.md (hunk 1 above)
Comment (avillela) [reply content/en/docs/kubernetes/operator/automatic.md:452]: I tried it, and it does crap out if you deploy in the wrong order.

**Review by TylerHelmuth — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Pull request merged**